lumi.bzl

```diff
@@ -0,0 +1,52 @@
+def _generated_c_impl(ctx):
+    args = [ctx.outputs.out.path]
+    args += [f.path for f in ctx.files.srcs]
+    args += [f.path for f in ctx.files.deps]
+
+    ctx.actions.run(
+    	inputs = ctx.files.srcs + ctx.files.deps,
+	outputs = [ctx.outputs.out],
+	executable = ctx.file.compiler,
+	arguments = args,
+	progress_message = "Generating {} from Lumi.".format(ctx.outputs.out.path),
+    )
+
+
+lumi_generated_c = rule(
+    attrs = {
+        "compiler": attr.label(
+	    default = Label("//TL5:compiler"),
+	    allow_single_file = True,
+	    executable = True,
+	    cfg = "host",
+	    doc = "Lumi compiler binary",
+	),
+	"srcs": attr.label_list(
+	    mandatory = True,
+	    allow_empty = False,
+	    allow_files = True,
+	    doc = "Lumi files to compile",
+	),
+	"deps": attr.label_list(),
+    },
+    implementation = _generated_c_impl,
+    outputs = {"out": "%{name}.c"},
+    output_to_genfiles = True,
+)
+
+
+def lumi_binary(name, srcs, deps, **kwargs):
```

Comment:
Currently Lumi may generate some unused variables and labels in the C code,
maybe add `-Wno-unused-label  -Wno-unused-variable` flags to the C compilation (and `-Wunused-but-set-variable` when using GCC)? so there will be no ugly warnings when compiling...

Reply:
also ` -Wno-missing-braces` (and `-Wno-self-assign` when using `clang`)

Reply:
This doesn't work for me on Windows with Blaze, for now. I think we can merge this and fix later if needed (it's a bit spammy, but it's not that bad).

Reply:
* Bazel